### DISCUSSION TITLE
Does WasmEdge support Tensorflow as its inference backend using WASI-NN?

### DISCUSSION THREAD

**Discussion by KeeganJin** [WasmEdge/WasmEdge#2765] [Q&A] [answered]
Hi all, 
I am working on a project to use a WASM runtime to perform inference. 
I checked the documents of WasmEdge and found that WasmEdge only supports Pytorch, OpenVINO, and Tensorflow-lite as its inference backend when using WASI-NN. Does that mean it does not support TensorFlow as its backend when using WASI-NN?

Thanks

**Comment by juntao** [comment]
That's correct. Tensorflow is not compatible with WASI NN. Can you export your Tensorflow model to Tensorflow lite? Thanks.

**Reply by KeeganJin** [reply]
Thanks for your reply. 
Sure it's possible to export the model to TensorFlow lite, and we actually would like to evaluate inference backends. 
I also checked other WASM runtimes, seems like all state-of-the-art standalone runtimes do not have the support for TensorFlow as a Backend with WASI NN?  Thank.

**Reply by juntao** [reply, marked as answer]
I think the problem is that tensorflow's API deign does not easily translate into WASI NN API. If you have a specific requirement for Tensorflow, you can checkout our Tensorflow plugin. 

https://wasmedge.org/docs/develop/rust/wasinn/tf_plugin

**Reply by KeeganJin** [reply]
Thanks a lot!
